src/snowflake/snowpark/column.py

```diff
@@ -993,12 +997,24 @@ def _create_ast(
         ast = proto.Expr()
         if property is not None:
```

Comment:
So where do we stand? Are you going to remove _create_ast()?

Reply:
Yes, we will remove `_create_ast()`, but we'll push this refactor to a separate PR in favor of narrowing this one's scope to just more informative error messages.